application/src/main/webapp/js/chat.js

```diff
@@ -830,51 +832,60 @@ var chatApplication = new ChatApplication();
         var $userResults = $("."+prefix+"-users-results");
         $userResults.css("display", "none");
         $userResults.html("");
-      } else {
-        chatApplication.getAllUsers(filter, function (jsonData) {
-          var users = TAFFY(jsonData.users);
-          var users = users();
-          var $userResults = $("."+prefix+"-users-results");
-          $userResults.css("display", "none");
-          var html = "";
-          if (!withCurrentUser) {
+      } else if (filter != searchingValue){
+        searchingValue = filter;
+        if (searchTimeout !== false) {
+          clearTimeout(searchTimeout);
+          searchTimeout = false;
```

Comment:
Seems set *searchTimeout = false* is useless. We don't even need the condition check *searchTimeout !== false*, always clearTimeout() before set a new one.